### ISSUE TITLE
the current cross_entropy_op does not support soft label.

### ISSUE THREAD

**Issue by lcy-seso** [PaddlePaddle/Paddle#3898]
The current [`cross_entropy_op`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/cross_entropy_op.h#L83) does not consider when a soft label is given rather than a hard label. This is not enough.

Here is the example in caffe2: https://github.com/caffe2/caffe2/blob/master/caffe2/operators/cross_entropy_op.cc#L200, and TF also supports the cross entropy with soft label.
